src/runtime/status.h

```diff
@@ -46,6 +57,12 @@ static inline tuple timm_internal(tuple t, char *first, ...)
         __up;                                       \
     })
 
+static inline void apply_status(status s, status_handler sh)
```

Comment:
OK, fair enough, I dropped this apply_status() function and reverted to just calling apply(). Since the timm_dealloc() function is not called anymore, it has been moved (along with destruct_tuple) to a separate commit (which is now mostly unrelated to this PR but doesn't seem to deserve a PR on its own).